From 1997b43f97544e509033c6d3b36faf1b74334892 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Mon, 6 Dec 2021 10:56:22 -0300
Subject: [PATCH 2/4] malloc trim

---
 bus/main.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/bus/main.c b/bus/main.c
index 5f756d5c..eb63cb7f 100644
--- a/bus/main.c
+++ b/bus/main.c
@@ -29,6 +29,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <malloc.h>
 #ifdef HAVE_SIGNAL_H
 #include <signal.h>
 #endif
@@ -747,6 +748,13 @@ main (int argc, char **argv)
 
   _dbus_verbose ("We are on D-Bus...\n");
   _dbus_daemon_report_ready ();
+  /*
+   * during init we may have allocated and then freed memory, which
+   * the C library will buffer for us. This is a good place to really
+   * free that memory. Any memory we allocate/free during our normal
+   * operation will still use the C library buffers.
+   */
+  malloc_trim(0);
   _dbus_loop_run (bus_context_get_loop (context));
 
   bus_context_shutdown (context);
-- 
2.34.0

