From 0df31ac8b24cad9ae54e44eeb55e11170d543f08 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Sat, 11 Dec 2021 19:25:08 -0300
Subject: [PATCH 3/5] DBUS_TEST_USER to boni

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0914f07b..9719aa20 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -627,7 +627,7 @@ else(WIN32)
     set(DBUS_SYSTEM_CONFIG_FILE  ${DBUS_DATADIR}/dbus-1/system.conf)
     set(DBUS_SESSION_CONFIG_FILE ${DBUS_DATADIR}/dbus-1/session.conf)
     set(DBUS_USER "messagebus")
-    set(DBUS_TEST_USER "nobody")
+    set(DBUS_TEST_USER "boni")
   # For best security, assume that all non-Windows platforms can do
   # credentials-passing.
     set(DBUS_SESSION_CONF_MAYBE_AUTH_EXTERNAL "<auth>EXTERNAL</auth>")
-- 
2.36.0

