From 11d04f30cf00cd04992516baaf0f25025be33144 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Sat, 11 Dec 2021 20:18:35 -0300
Subject: [PATCH 5/5] unset DBUS_FATAL_WARNINGS DBUS_SESSION_BUS_ADDRESS

---
 cmake/modules/Macros.cmake | 2 --
 1 file changed, 2 deletions(-)

diff --git a/cmake/modules/Macros.cmake b/cmake/modules/Macros.cmake
index 92fe9d81..8eb4308a 100644
--- a/cmake/modules/Macros.cmake
+++ b/cmake/modules/Macros.cmake
@@ -57,8 +57,6 @@ macro(add_unit_test _name _target)
         WORKING_DIRECTORY ${DBUS_TEST_WORKING_DIR}
     )
     set(_env)
-    list(APPEND _env "DBUS_SESSION_BUS_ADDRESS=")
-    list(APPEND _env "DBUS_FATAL_WARNINGS=1")
     list(APPEND _env "DBUS_TEST_DAEMON=${DBUS_TEST_DAEMON}")
     list(APPEND _env "DBUS_TEST_DATA=${DBUS_TEST_DATA}")
     list(APPEND _env "DBUS_TEST_DBUS_LAUNCH=${DBUS_TEST_DBUS_LAUNCH}")
-- 
2.34.1

